pavelib/quality.py

```diff
@@ -756,27 +756,68 @@ def _get_xsscommitlint_count(filename):
         return None
 
 
+def _extract_missing_pii_annotations(filename):
+    """
+    Returns the number of uncovered models from the stdout report of django_find_annotations.
+
+    Arguments:
+        filename: Filename where stdout of django_find_annotations was captured.
+
+    Returns:
+        Number of uncovered models.
+    """
+    uncovered_models = None
+    if os.path.isfile(filename):
+        with open(filename, 'r') as report_file:
+            lines = report_file.readlines()
+            uncovered_regex = re.compile(r'^Coverage found ([\d]+) uncovered')
+            for line in lines:
+                uncovered_match = uncovered_regex.match(line)
+                if uncovered_match:
+                    uncovered_models = int(uncovered_match.groups()[0])
+                    break
+    else:
+        fail_quality('pii', u'FAILURE: Log file could not be found: {}'.format(filename))
+
+    return uncovered_models
+
+
 @task
 @needs('pavelib.prereqs.install_python_prereqs')
+@cmdopts([
+    ("report-dir=", "r", "Directory in which to put PII reports"),
+])
 @timed
 def run_pii_check(options):  # pylint: disable=unused-argument
     """
     Guarantee that all Django models are PII-annotated.
     """
+    pii_report_name = 'pii'
+    default_report_dir = (Env.REPORT_DIR / pii_report_name)
+    report_dir = getattr(options, 'report_dir', default_report_dir)
+    output_file = os.path.join(report_dir, 'pii_check_{}.report')
+    uncovered_model_counts = []
     for env_name, env_settings_file in (("CMS", "cms.envs.test"), ("LMS", "lms.envs.test")):
         try:
             print()
             print("Running {} PII Annotation check and report".format(env_name))
             print("-" * 45)
+            run_output_file = six.text_type(output_file).format(env_name.lower())
             sh(
                 "export DJANGO_SETTINGS_MODULE={}; "
                 "code_annotations django_find_annotations "
-                "--config_file .pii_annotations.yml --report_path pii_report/ "
-                "--lint --report --coverage".format(env_settings_file)
+                "--config_file .pii_annotations.yml --report_path {} "
+                "--lint --report --coverage | tee {}".format(env_settings_file, report_dir, run_output_file)
```

Comment:
This appears to create the following console logs:

* `pii/pii_check_cms.report`
* `pii/pii_check_lms.report`

but the PII report files will be:

* `pii/yyyy-dd-mm-hh-mm-ss.yaml`
* `pii/yyyy-dd-mm-hh-mm-ss.yaml`

where the timestamp will just be slightly different between LMS and CMS.  Would it be better to also call out the lowercase env_name in the report filename too?  Is that not possible with our current code-annotations tooling?

Reply:
I've got a code_annotations PR brewing for this: https://github.com/edx/code-annotations/pull/28